cfme/tests/infrastructure/test_host.py

```diff
@@ -553,16 +554,36 @@ def test_infrastructure_hosts_crud(appliance, setup_provider_min_hosts, provider
     Bugzilla:
         1634794
     """
-    my_slice = slice(0, num_hosts, None)
-    hosts_view = navigate_to(provider.collections.hosts, "All")
-    for h in hosts_view.entities.get_all(slice=my_slice):
-        h.ensure_checked()
-    hosts_view.toolbar.configuration.item_select('Edit Selected items',
-                                                handle_alert=False)
-    try:
-        hosts_view.navigation.select('Compute', 'Infrastructure', 'Providers', handle_alert=False)
-        final_view = provider.create_view(InfraProvidersView)
-        assert final_view.is_displayed
-    except UnexpectedAlertPresentException:
-        pytest.fail("Abandon changes alert displayed, but no changes made.")
-    # TODO add additional crud functionality.
+    host = appliance.collections.hosts.all()[0]
+    if crud_action in ['cancel', 'nav_away_changes', 'nav_away_no_changes']:
+        try:
+            existing_custom_id = navigate_to(host, 'Details').entities.summary(
+                "Properties").get_text_of("Custom Identifier")
+        except NameError:
+            existing_custom_id = None
+    if crud_action != 'remove':
+        stamp = fauxfactory.gen_alphanumeric()
+        new_custom_id = f'Edit host data. {stamp}'
+        try:
+            with update(host, action=crud_action):
+                host.custom_ident = new_custom_id
+        except UnexpectedAlertPresentException as e:
```

Comment:
I am puzzled. Do you mean to investigate the exception only if it happens? And if it happens only if the action is some specific one? This doesn't seem right.

If you expect the action to raise an exception, you should do something like this:
```
if crud_action in ['cancel', 'nav_away_no_changes']:
    with pytest.raises(UnexpectedAlertPresentException) match=r".*Abandon changes.*"):
        with update(host, action=crud_action):
            host.custom_ident = new_custom_id
```

The other catches are perhaps also candidates to be done like that.

Optional: I would think about a way to reduce the amount of branches. Perhaps something like I did in https://github.com/ManageIQ/integration_tests/pull/10019/files#diff-fc91ed43845421f3eda7a5bb17217dc7R279

Reply:
This is to catch an exception that was thrown before the fix for BZ 1634794 was merged. This has been fixed but I want to catch it if it occurs again.

Reply:
What does catching and re-raising exceptions accomplish here and in L583-L588? If `update()` raises an exception, then pytest logs it and the test fails, regardless.

Reply:
I was just trying to catch a specific exception that was caused by an old BZ. Maybe I should have mentioned the old BZ in the logging test. Basically, I just wanted it to be easier to tell that the BZ this test is created to catch is reoccurring when/if it does.  If this is something we really discourage, I can remove it but I think it makes the failure clearer.

Reply:
I added BZ number to log message. More input on this?

Reply:
I think its a good thing to catch this and fail explicitly with a prior BZ